src/rpp/jni_cpp.cpp

```diff
@@ -52,6 +52,14 @@ namespace rpp { namespace jni {
         return androidActivity;
     }
 
+    void initMainActivity(jobject globalHandle)

```

Comment:
You should also update the `getActivity()` to try multiple fields:
```cpp
if (auto currentActivity =  mainClass.SField("currentActivity", "Landroid/app/Activity;", std::nothrow)
{
    androidActivity = currentActivity.Object().to_global();
}
else if (auto activity = ......

```

Reply:
I guess there is a mistake in the `nothrow` path. The exception needs to be cleared on failure.